yellowbrick/features/scatter.py

```diff
@@ -0,0 +1,278 @@
+
+        :param features: a list of feature names to use
+            List of features that correspond to the columns in the array.
+            More than two feature names or columns will raise an error. If
+            a DataFrame is passed to fit and features is None, feature
+            names are selected that are the columns of the DataFrame.
+
+        :param classes: a list of class names for the legend
+            If classes is None and a y value is passed to fit then the classes
+            are selected from the target vector.
+
+        :param color: optional list or tuple of colors to colorize points
+            Use either color to colorize the points on a per class basis or
+            colormap to color them on a continuous scale.
+
+        :param colormap: optional string or matplotlib cmap to colorize points
+            Use either color to colorize the points on a per class basis or
+            colormap to color them on a continuous scale.
+
+        :param markers: iterable of strings
+            Matplotlib style markers for points on the scatter plot points
+
+        :param kwargs: keyword arguments passed to the super class.
+
+        These parameters can be influenced later on in the visualization
+        process, but can and should be set as early as possible.
+        """
+        super(ScatterVisualizer, self).__init__(ax, features, classes, color,
+                                                colormap, **kwargs)
+        self.markers = itertools.cycle(
+            kwargs.pop('markers', (',', '+', 'o', '*', 'v', 'h', 'd')))
+
+        # Ensure with init that features doesn't have more than two features
+        if features is not None:
+            if len(features) != 2:
+                raise YellowbrickValueError(
+                    'ScatterVisualizer only accepts two features')
+
+    def fit(self, X, y=None, **kwargs):
+        """
+        The fit method is the primary drawing input for the parallel coords
+        visualization since it has both the X and y data required for the
+        viz and the transform method does not.
+
+        Parameters
+        ----------
+        X : ndarray or DataFrame of shape n x m
+            A matrix of n instances with 2 features
+
+        y : ndarray or Series of length n
+            An array or series of target or class values
+
+        kwargs : dict
+            Pass generic arguments to the drawing method
+
+        Returns
+        -------
+        self : instance
+            Returns the instance of the transformer/visualizer
+        """
+        _, ncols = X.shape
+
+        if ncols == 2:
+            X_two_cols = X
+            if self.features_ is None:
+                self.features_ = [0, 1]
+
+        # Handle the feature names if they're None.
+        elif self.features_ is not None and is_dataframe(X):
+            X_two_cols = X[self.features_].as_matrix()
+        else:
+            raise YellowbrickValueError("""
```

Comment:
I'll add the grabbing features by via indexes of narrays without names.

I added in `is_structured_array` similar to is_dataframe, and came across an interesting internal mplt error. 

`TypeError: ufunc 'invert' not supported for the input types, and the inputs could not be safely coerced to any supported types according to the casting rule ''safe''`

Take another crack at this tomorrow.